### PR TITLE
Merge 8.2 final into trunk

### PR DESCRIPTION
**Pull request by jkmassel**
Merges the 8.2 finalization into `trunk` (note – `trunk` has replaced `develop`).

### CODE REVIEW

**Review by oguzkocer — APPROVED** [review]

**Comment by mokagio** [comment]
🤔 the [tests failed](https://app.circleci.com/pipelines/github/woocommerce/woocommerce-android/17049/workflows/d1f5d34b-89b5-45ee-a95b-e1e8908fee94/jobs/59368) with:

- `java.lang.AssertionError: expected:<Thu Jul 01 00:00:00 UTC 2021> but was:<Thu Oct 01 00:00:00 UTC 2020>`
- `java.lang.AssertionError: expected:<Fri Oct 01 00:00:00 UTC 2021> but was:<Fri Jan 01 00:00:00 UTC 2021>`

https://github.com/woocommerce/woocommerce-android/pull/5589 should solve the issue. From what I understand of the code, that was an issue that could occur only in test code or code that would otherwise pass a `Calendar` value different from the default (`Calendar.getInstance()`). As such, I don't think the tests failing here mean the app would be showing the wrong quarter value to the users.

@wzieba you worked on the fix and might have more context, could you confirm? Thanks!

**Comment by wzieba** [comment]
Thanks @mokagio for ping!

> From what I understand of the code, that was an issue that could occur only in test code or code that would otherwise pass a `Calendar` value different from the default (`Calendar.getInstance()`).

Yes, in production code those two methods that failed tests are using default argument of `Calendar.getInstance()`.

> As such, I don't think the tests failing here mean the app would be showing the wrong quarter value to the users.

Exactly, it was just a test issue. _If_ we'll spot wrongly calculated quarter, it'd be because of flaw in logic, not passing wrong parameter. So in summary #5589 only fixed failing tests.

**Comment by mokagio** [comment]
Thanks for the confirmation and for the fix @wzieba. 💜 🙇 

@jkmassel did enable auto-merge for this, but it didn't trigger because of the failing tests. Given we established the failing tests are not a sign of an issue in the production experience, I will merge this manually now to keep the process flowing.

**Comment by mokagio** [comment]
> Given we established the failing tests are not a sign of an issue in the production experience, I will merge this manually now to keep the process flowing.

Turns out I cannot use my admin privileges to merge because the branch protection rules on `trunk` do not allow it 🤔 

I'm going to cherry pick #5589 to work around it as we don't like to make one-off exceptions to those rules without discussing first.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/util/DateUtils.kt [hunk 1]
```diff
@@ -435,7 +435,7 @@ class DateUtils @Inject constructor(
             clear(Calendar.MINUTE)
             set(Calendar.HOUR_OF_DAY, ZERO)
             set(Calendar.DAY_OF_MONTH, 1)
-            set(Calendar.MONTH, Calendar.getInstance().get(Calendar.MONTH) / THREE * THREE)
+            set(Calendar.MONTH, get(Calendar.MONTH) / THREE * THREE)
```
Comment (mokagio) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/util/DateUtils.kt:438]: I'm not sure why the changes to this file appear in the diff, when the commit that made the is already on `trunk`, and you can also see the code as here by exploring in the GitHub UI:

https://github.com/woocommerce/woocommerce-android/blob/65c68bdec12ed041fcb4d569455fdf984c81534f/WooCommerce/src/main/kotlin/com/woocommerce/android/util/DateUtils.kt#L382

https://github.com/woocommerce/woocommerce-android/blob/65c68bdec12ed041fcb4d569455fdf984c81534f/WooCommerce/src/main/kotlin/com/woocommerce/android/util/DateUtils.kt#L438

**Comment by jkmassel** [comment]
Thanks for taking care of landing this safely @mokagio!!

**Pull request merged**